src/chunk-stream.iced

```diff
@@ -1,17 +1,26 @@
 stream = require('stream')
+{make_esc} = require('iced-error')
 
 exports.ChunkStream = class ChunkStream extends stream.Transform
 
-  constructor : ({@transform_func, @block_size, @exact_chunking, @writableObjectMode, @readableObjectMode}) ->
+  constructor : ({@transform_func, @block_size, @readableObjectMode}) ->
     @extra = null
-    super({@writableObjectMode, @readableObjectMode})
+    super({@readableObjectMode})
+
+  # chunk must be guaranteed to be a multiple of the block size
+  _transform_chunk : (chunk, cb) ->
+    blocks = []
+    await
```

Comment:
see ec02721 - it was `Buffer.concat` that was the bottleneck (we're now doing one big `concat` on the `blocks` array, instead of `concat`ing each block as we go along). I tested parallel vs serial with the same concat-an-array scheme and found no difference, so it's serial now for simplicity.